provisioning/containers/gui-common/websockify/novnc-overrides/app/ui.js

```diff
@@ -1,34 +1,223 @@
+            this.cpu = (((this.cpu * this.historyLen) - this.lastResources[this.currentIndex].cpu + resources.cpu) / this.historyLen) || 0;
+            this.mem = (((this.mem * this.historyLen) - this.lastResources[this.currentIndex].mem + resources.mem) / this.historyLen) || 0;
+            this.net = (((this.net * this.historyLen) - this.lastResources[this.currentIndex].net + resources.net) / this.historyLen) || 0;
+        }
+        this.lastResources[this.currentIndex % this.historyLen] = resources;
+        this.currentIndex = (this.currentIndex + 1) % this.historyLen;
+    }
+}
+
+var resourcesMean = new ResourcesHistory();
+var currentResources = new Resources(null, null, null, null, null, "No fetched resources available");
+
+/**
+ * Register actions needed when page is loaded:
+ *  - WS metrics connection
+ *  - Button drag and drop logic
+ */
+window.addEventListener('load', () => {
+    // WS metrics connection
+    var metricsPath = "ws://" + document.location.host + "/usages?updatePeriod=2";
+    if (window.connUid != undefined) {
+        metricsPath = `${metricsPath}&connUid=${window.connUid}`;
+    }
+    const conn = new WebSocket(metricsPath);
+    conn.onclose = (evt) => {
+        document.getElementById("usage-btn-txt").innerHTML = 'WS Closed';
+        Log.Error("Usages WS connection closed: ", evt);
+    }
+    conn.onerror = (err) => {
+        document.getElementById("usage-btn-txt").innerHTML = 'WS Error';
+        Log.Error("Usages WS connection error: " + err);
+    }
+    conn.onmessage = (evt) => {
+        try {
+            const resourcesJson = JSON.parse(evt.data.toString());
+            currentResources = Resources.from(resourcesJson);
+            resourcesMean.addResources(currentResources);
+            updateViewUsages();
+        } catch (error) {
+            Log.Error("Error on JSON parsing from WS data: ", error);
+            document.getElementById("usage-btn-txt").innerHTML = 'WS Error';
+        }
+    };
+});
+
+/**
+ * Register actions needed when DOM content is loaded:
+ *  - Button drag and drop logic
+ */
+document.addEventListener('DOMContentLoaded', () => {
+    const button = document.getElementById('usage-btn')
+    var isMoving = false;
+    var initialX, initialY;
+
+    button.addEventListener('mousedown', (e) => {
+        isMoving = true;
+        initialX = e.clientX - button.offsetLeft;
+        initialY = e.clientY - button.offsetTop;
+
+        document.body.insertAdjacentHTML('beforeend', `<div id="page-element" class="dim-screen"></div>`);
+        document.getElementById('page-element').addEventListener('mousemove', (e) => {
+            if (isMoving) {
+                var { clientX, clientY } = e;
+
+                if (clientX < 30) clientX = 30;
+                if (clientX > window.innerWidth - 60) clientX = window.innerWidth - 60;
+                if (clientY < 200) clientY = 200;
+                if (clientY > window.innerHeight - 30) clientY = window.innerHeight - 30;
```

Comment:
Try not to use fixed values, if possible it would be better to work with css in order to make these constant values not required